lightning/src/ln/channel.rs

```diff
@@ -1136,9 +1150,7 @@ impl<Signer: Sign> Channel<Signer> {
 		} else {
 			ChannelTypeFeatures::from_counterparty_init(&their_features)
 		};
-		if !channel_type.supports_static_remote_key() {
-			return Err(ChannelError::Close("Channel Type was not understood - we require static remote key".to_owned()));
-		}
+		let opt_anchors = channel_type.supports_anchors_zero_fee_htlc_tx();
```

Comment:
The test case which could be valuable would be actually someone feeding opening/accepting messages with `option_anchor_outputs` and check we either drop the funding flow or overrule to the safe `option_anchor_outputs`. Being robust against channel type safe downgrade attempts.